app/models/contractor/contractor.rb

```diff
@@ -31,17 +31,12 @@ class DiaperDriveParticipant < ApplicationRecord
   after_validation :geocode, if: ->(obj) { obj.address.present? && obj.address_changed? }
 
```

Comment:
We should move the `phone, email` validation to the diaper drive participant class. It doesn't make sense to them to put in that information on all their vendors. For example, they might have amazon.com as a vendor and they probably won't have a phone and email for it.

Reply:
I'm not sure I understand the decision to use the word `Contractor` for this supertype.

The first word that comes to mind for me would be `Provider` -- please push back if you have a good reason for `Contractor` though -- maybe I'm not seeing something. 

I just want to make sure we're using language that is intuitive -- we've (..."I've"...) made a couple decisions in the past to use strange words for models (most have been changed, thankfully), and it's created confusion

Reply:
I checked russian-english vocabulary. :) Synonims of this word are "contractor; counteragent; partner; contracting party; covenantee, covenantor".
You are a native speaker, so if you like Partner, i'll rename it.